web/src/components/organisms/Project/ModelsMenu/hooks.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  useGetModelsQuery,
+  useCreateModelMutation,
+  useCheckModelKeyAvailabilityLazyQuery,
+} from "@reearth-cms/gql/graphql-client-api";
+
+type Params = {
+  projectId?: string;
+  modelId?: string;
+};
+
+export default ({ projectId, modelId }: Params) => {
+  const [modelModalShown, setModelModalShown] = useState(false);
+  const [isKeyAvailable, setIsKeyAvailable] = useState(false);
+  const [CheckModelKeyAvailability, { data: keyData }] = useCheckModelKeyAvailabilityLazyQuery({
+    fetchPolicy: "no-cache",
+  });
+
+  const handleModelKeyCheck = useCallback(
+    async (projectId: string, key: string) => {
+      if (!projectId || !key) return false;
+      const response = await CheckModelKeyAvailability({ variables: { projectId, key } });
+      return response.data ? response.data.checkModelKeyAvailability.available : false;
+    },
+    [CheckModelKeyAvailability],
+  );
+
+  useEffect(() => {
+    setIsKeyAvailable(keyData?.checkModelKeyAvailability.available ?? false);
+  }, [keyData?.checkModelKeyAvailability]);
+
+  const { data } = useGetModelsQuery({
+    variables: { projectId: projectId ?? "", first: 100 },
+    skip: !projectId,
+  });
+
+  const models = useMemo(() => {
+    return (data?.models.nodes ?? [])
+      .map<Model | undefined>(model =>
+        model
+          ? {
+              id: model.id,
+              description: model.description,
+              name: model.name,
+              key: model.key,
+              schema: {
+                id: model.schema.id,
+                fields: model.schema.fields.map(field => ({
+                  id: field.id,
+                  description: field.description,
+                  title: field.title,
+                  type: field.type,
+                  key: field.key,
+                  unique: field.unique,
+                  required: field.required,
+                  typeProperty: field.typeProperty,
+                })),
+              },
+            }
+          : undefined,
+      )
+      .filter((model): model is Model => !!model);
+  }, [data?.models.nodes]);
+
+  const rawModel = useMemo(
+    () => data?.models.nodes.find((p: any) => p?.id === modelId),
+    [data, modelId],
+  );
+  const model = useMemo<Model | undefined>(
+    () =>
+      rawModel?.id
+        ? {
+            id: rawModel.id,
+            description: rawModel.description,
+            name: rawModel.name,
+            key: rawModel.key,
+            schema: {
+              id: rawModel.schema.id,
+              fields: rawModel.schema.fields.map(field => ({
+                id: field.id,
+                description: field.description,
+                title: field.title,
+                type: field.type,
+                key: field.key,
+                unique: field.unique,
+                required: field.required,
+                typeProperty: field.typeProperty,
+              })),
+            },
+          }
+        : undefined,
```

Comment:
```suggestion
      rawModel?.id ? fromModel(rawModel) : undefined,
```
Just a suggestion, please let me know what you think!
And feel free to ignore it.

Reply:
Done.
https://github.com/reearth/reearth-cms/blob/14b37f7e299079a1014ea055b0000128d996e7b1/web/src/components/organisms/Project/ModelsMenu/hooks.ts#L41-L70